packages/hardhat-core/src/internal/hardhat-network/provider/modules/eth.ts

```diff
@@ -1272,7 +1272,8 @@ export class EthModule {
       return;
     }
 
-    const code = await this._node.getCode(trace.address, "pending");
```

Comment:
@duckception check this out, this was the other line that caused eth_call to always mine a pending block